### PR TITLE
feat: proof of concept to inject react components in md files

### LINKED ISSUE #34494: Move Coding Bootcamp Cost Calculator to the Guide
I believe with Gatsby we can just have an interactive Guide article instead of having a JavaScript app that lives at a specific path. Then we can just create a 301 from https://www.freecodecamp.org/coding-bootcamp-cost-calculator to freecodecamp.org/guide/tools/calculators/coding-bootcamp-cost-calculator

This would be an important proof of concept we could use to create other interactive calculators.

For example, our Binary Calculator article could become more like this: https://www.calculator.net/binary-calculator.html

### PR DESCRIPTION
**Pull request by ronaldcs**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #34494

### CODE REVIEW

**Review by ValeraS — CHANGES_REQUESTED** [review]
It's a working solution, but I think it does not meet our needs. It's good for cases where you have the same few components on many pages. But we have unique components on pages.

I propose next:
- in `index.md` add into frontmatter line:
```yml
component: calculators/CodingBootcampCodeCalculator
```
- place component files to `src/templates/Guide/components/calculators/CodingBootcampCodeCalculator`
- set `index.js` name for main file with default export and add a guide article layout to it.
- in `gatsby-node.js` add `component` to graphQL query
- in `utils/gattsby/guidePageCreator.js` 
```diff
@@ -32,7 +33,14 @@ exports.createGuideArticlePages = createPage => ({

   return createPage({
     path: `/guide${slug}`,
-    component: guideArticle,
+    component: !component
+      ? guideArticle
+      : path.resolve(
+          __dirname,
+          '../../src/templates/Guide/components/',
+          component,
+          'index.js'
+        ),
     context: {
       id,
       meta
```

if @Bouncey agree then I'll make proposed changes and @ronaldcs change CodingBootcampCodeCalculator and `index.md` files.

File: client/src/components/Calculators/CodingBootcampCostCalculator.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import { initCalculator, node } from './calculator';
+import './codingbootcampcostcalculator.css';
+
+const RD3Component = rd3.Component;
+
+export default class CodingBootcampCostCalculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      d3: node,
```
[the remark this answers is not in the document]
Comment (ronaldcs) [reply client/src/components/Calculators/CodingBootcampCostCalculator.js]: Seems like the only way I could get this to work was to use `document.createElement('div')` to get it to play nice with react-d3-library. I've tried `createRef` but that didn't work. Suggestions welcome.

File: client/src/components/Calculators/CodingBootcampCostCalculator.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import rd3 from 'react-d3-library';
```
[the remark this answers is not in the document]
Comment (ronaldcs) [reply client/src/components/Calculators/CodingBootcampCostCalculator.js]: Excellent, thanks for the lesson :-).

File: client/src/components/Calculators/CodingBootcampCostCalculator.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import CostCalculator from './CostCalculator';
+import './codingbootcampcostcalculator.css';
+
+class CodingBootcampCostCalculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      d3: <div />,
```
Comment (ronaldcs) [inline client/src/components/Calculators/CodingBootcampCostCalculator.js]: Not needed anymore, will remove.

**Comment by ValeraS** [comment]
@ronaldcs Could you redo this component like it done in this example https://github.com/freeCodeCamp/freeCodeCamp/commit/5a8f9bff4134519b477eb7db85940597c3f8a78b

**Comment by ronaldcs** [comment]
@ValeraS sure, I can work on that.

**Comment by ronaldcs** [comment]
https://github.com/freeCodeCamp/freeCodeCamp/pull/35278 replaces this PR relative to moving bootcamp cost calculator to guide.

**Pull request closed without merging**